### DISCUSSION TITLE
Template for teaching purposes

### DISCUSSION THREAD

**Discussion by jmgirard** [quarto-dev/quarto-cli#5940] [Q&A] [answered]
### Description

When teaching, I need students to render their QMD files to HTML with resources embedded so that they can easily share these files with me without the formatting breaking. However, it is a pain to have to teach them about YAML headers at the very beginning. I would rather circle back to this later. I think a good solution for this would be something like a Quarto version of RNotebooks - a file type or format that automatically embeds resources and perhaps turns on paged df printing. Ideally, there would be a way to create such a file through the RStudio IDE directly (e.g., either a different file type or a checkbox on the Quarto document wizard).

**Comment by mcanouil** [comment]
You are describing a format template extension, see <https://quarto.org/docs/extensions/formats.html>.

Then your students, will only have to run `quarto use template <gh-organization>/<extension>` (<https://quarto.org/docs/extensions/formats.html>) and start working on the Quarto document.

**Reply by jmgirard** [reply]
This is a good start, but I'd ideally like it to be built into RStudio somehow. Teaching them quarto CLI commands is on a similar level of technicalness as customizing the YAML header. Do you know if it is possible to add new file types through an RStudio addon?

**Reply by cscheid** [reply]
That would be a question for https://github.com/rstudio/rstudio or their discussion forums.

**Reply by mcanouil** [reply, marked as answer]
See #2079 for related if not identical discussion.

**Reply by jmgirard** [reply]
Thanks @mcanouil that seems super relevant.

**Reply by mcanouil** [reply]
FWIW, I don't think asking your students to copy paste a quarto command on the terminal within RStudio is "too advanced".
They are running commands in R 😅

**Reply by jmgirard** [reply]
As surprising as it might seem, this is a barrier to entry for some newbies. I am trying to bring people with zero programming experience into R. They are just wrapping their minds around the R console and it would be confusing and error-prone to introduce another, secondary console (i.e., the terminal) with different commands just to use it once and never again.

**Reply by mcanouil** [reply]
I see.

Even if you had a button in the RStudio interface, it would require them to install an R package, likely from GitHub, thus using non R base function, etc.
In the end, I am not sure it is less error prone. 🤷‍♂️

**Reply by jmgirard** [reply]
I agree. Ideally, a quarto version of R Notebooks would be built into RStudio directly. But barring that possibility, getting such a package into CRAN might do it.

**Reply by mcanouil** [reply]
What you call "notebook" is unlikely to happen.
For instance all template in Rmkardown are third parties R Packages.
RStudio is simply looking for a particular directory in the R packages.

Nothing is defined in RStudio for where to look for such files in R packages.

As answered, this is a feature for RStudio IDE (and Quarto VSCode extension, see <https://github.com/quarto-dev/quarto>).

**Reply by mcanouil** [reply]
@jmgirard Do you mind if I move this to "Q&A" instead of "Feature Requests"?

**Reply by jmgirard** [reply]
Yes, move to  Q&A. I'll request this from the RStudio IDE.

**Reply by mcanouil** [reply]
Done, feel free to mark as the answer that you feel answer the best the question. 😉

**Reply by cderv** [reply]
Just for reference, linking to the related issue on RStudio IDE repo: https://github.com/rstudio/rstudio/issues/13256
